src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1529,6 +1540,27 @@ internal static SqlConnection OpenSqlConnection(ConnectionInfo connInfo, string
             return null;
         }
 
+        /// <summary>
+        /// Create and open a new ServerConnection from a ConnectionInfo object.
+        /// This calls ConnectionService.OpenSqlConnection and then creates a
+        /// ServerConnection from it.
+        /// </summary>
+        internal static ServerConnection OpenServerConnection(ConnectionInfo connInfo, string featureName = null)

```

Comment:
it would be good to add some description in doc comments about the 'featureName' parameter. It is not obvious how that parameter is used or what it does.

Reply:
done